app/models/spree/gateway/amazon.rb

```diff
@@ -86,11 +92,16 @@ def capture(amount, amazon_checkout, gateway_options={})
       if amount < 0
         return credit(amount.abs, nil, nil, gateway_options)
       end
-      order = Spree::Order.find_by(:number => gateway_options[:order_id].split("-")[0])
+      order_number, payment_number = gateway_options[:order_id].split("-", 2)
+      order = Spree::Order.find_by!(number: order_number)
+      payment = Spree::Payment.find_by!(number: payment_number)
+      authorization_id = order.amazon_transaction.authorization_id
+      capture_reference_id = "#{payment.number}-#{random_suffix}"
```

Comment:
Yah, I've added a bit of explanation to the code with this comment:

> Amazon requires unique ids. Calling with the same id multiple times means
> the result of the previous call will be returned again. This can be good
> for things like asynchronous retries, but would break things like multiple
> captures on a single authorization.

So maybe we could figure out how to take advantage of this, but as-is I don't think it's completely safe to re-use the same unique id.